sys/include/vfs.h

```diff
@@ -64,7 +64,11 @@
 #endif
 #include <sys/stat.h> /* for struct stat */
 #include <sys/types.h> /* for off_t etc. */
-#include <sys/statvfs.h> /* for struct statvfs */
+
+#ifdef CPU_NATIVE
+/* on native, use system's struct statvfs definition */
```

Comment:
This IMO also supports @chrysn assessment that we shouldn't reuse POSIX types for RIOT's native non-POSIX VFS API.